### DISCUSSION TITLE
Mouse is able to clic but not to move

### DISCUSSION THREAD

**Discussion by l-lalex** [tiny-pilot/tinypilot#541] [Q&A]
Hi,
 I tried tinypilot with a Dahua NVR because the recorder's web  frontend is very poor instead of KVM.
But the remote mouse does not move. However, the remote clics right and left are functional.
I tried with a connected mouse both (and not both) with tinypilot and the connected one is OK in all test but not the remote one.
I tried tinypilot with a PC and then the remote mouse moving is OK.
Have you idea how to fix this ?

**Comment by mtlynch** [comment]
Sorry to hear this!

What model number system are you using?

TinyPilot presents itself to the target system as a USB hub with a USB keyboard and mouse inserted. For some systems like KVMs and NVRs, they're expecting a simple USB keyboard or mouse rather than a hub, so they don't accept the input. That's likely what's happening here.

**Reply by l-lalex** [reply]
I use a DHI-NVR4108-P-4KS2 with the latest tinypilot version.
This appliance have 2 USB ports.
How can I find in logs that you are explaining ?
Is exist a way to fix this ?

**Reply by mtlynch** [reply]
So mouse *movement* works but it doesn't pick up clicks?

>How can I find in logs that you are explaining ?

Can you [update to the latest version](https://github.com/mtlynch/tinypilot#updates) and then from the top navbar, go to System > View Logs?

![image](https://user-images.githubusercontent.com/7783288/110390117-7d5f0a00-8033-11eb-833b-4a2a0c173f23.png)

**Reply by l-lalex** [reply]
mouse is clicking (not 100% of tests) but not moving.
Nice job for the logs.
Do you think il is possible to fix the issue ?
https://logs.tinypilotkvm.com/Cwz4gk5g

**Reply by mtlynch** [reply]
Hmm, the logs don't show any errors passing through the mouse movement.

My guess is that the NVR doesn't recognize TinyPilot's type of mouse. TinyPilot presents itself to the target computer as an absolute-position mouse ("go to position 250px, 500px," like a trackpad), but mice are more commonly relative-position (e.g., "move 200px right from current position"). The NVR might only work with relative-position mice.

There unfortunately isn't a fix unless you want to [edit the HID configuration](https://github.com/mtlynch/ansible-role-tinypilot/blob/d12fe2a2fbb4d7e7150252ab9a94fce10fa50100/files/init-usb-gadget#L76L123) for the mouse and try with a relative-position HID, but that's a complicated task.
